app/fr/maif/izanami/utils/controllerHelpers.scala

```diff
@@ -0,0 +1,17 @@
+package fr.maif.izanami.utils
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.Results.BadRequest
+
+import scala.concurrent.Future
+
+object ControllerHelpers {
+   def checkPassword(body: JsValue): Future[Either[play.api.mvc.Result, String]] = {
```

Comment:
Actual password checking should be factorized as well, moreover this should probably be done in an Action. I think it may be worth taking a look at [this doc page](https://www.playframework.com/documentation/3.0.x/ScalaActionsComposition) to see how we could combine two actions (previous controller action and this new password verification action).

Reply:
I will have a look on the documentation and modify it, shall I keep the ControllerHelpers or I need to remove it too ?

Reply:
You should put it in [AuthAction](https://github.com/MAIF/izanami/blob/main/app/fr/maif/izanami/web/AuthAction.scala)